K2Bridge/Visitors/Aggregations/RangeAggregationVisitor.cs

```diff
@@ -19,48 +20,66 @@ public void Visit(RangeAggregation rangeAggregation)
             EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Metric, nameof(RangeAggregation.Metric));
             EnsureClause.StringIsNotNullOrEmpty(rangeAggregation.Field, nameof(RangeAggregation.Field));
 
-            // Start the union operator
-            rangeAggregation.KustoQL += $"{KustoQLOperators.Union} ";
-
-            // Query expressions for each range
-            // (_data | where foo >= 1 and bar < 10 | summarize ['3']=avg(baz), count() | extend ['2'] = '1-10')
-            //          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
-            //          The range clause
-            //                                        ^^^^^^^^^^^^^^^^^^^^^^^^
-            //                                        The metrics expressions
-            //                                                                  ^^^^^^^
-            //                                                                  Default metric count()
-            //                                                                            ^^^^^^^^^^^^^^^^^^^^^
-            //                                                                            Column with the range name
+            string expandColumn = EncodeKustoField("_range_value");
+
+            // Extend expression:
+            // >> ['2']=pack_array("range1", "range2", "range3"), ['_range_value'] = pack_array(expr1, expr2, expr3)
+            // >> | mv-expand ['2'] to typeof(string), ['_range_value']
+            // >> | where ['_range_value'] == true
+            var extendExpression = new StringBuilder();
+
+            var rangeNames = new List<string>();
```

Comment:
same comment for range expression?

Reply:
Yes. RangeAggregationVisitor and DateRangeAggregationVisitor are similar. Difference is between RangeAggregationExpressionVisitor and DateRangeAggregationExpressionVisitor.

Reply:
I meant, can you add a similar comment like you added above to the extendExpression.

// Range expression:
            // >> abc...